### ISSUE TITLE
Trees should have terminal nodes

### ISSUE THREAD

**Issue by tercer** [TwoRavens/TwoRavens#145]
The trees have decision nodes as the final leaves/terminal nodes.  However, you can't read from that structure how much data is separated by that final decision node.  Also, you can't see what the final prediction is for the observation when you've traced it's path through the tree.

For example, if the variable is _war years_ the final node should be a probability or mean of the observations that reach each end point.

In summary, the trees should be one node longer to be more useful.
